common/djangoapps/student/views/dashboard.py

```diff
@@ -848,7 +863,22 @@ def student_dashboard(request):  # lint-amnesty, pylint: disable=too-many-statem
         'resume_button_urls': resume_button_urls
     })
 
-    response = render_to_response('dashboard.html', context)
+    dashboard_template = 'dashboard.html'
+    try:
+        # .. filter_implemented_name: DashboardRenderStarted
+        # .. filter_type: org.openedx.learning.dashboard.render.started.v1
+        context, dashboard_template = DashboardRenderStarted.run_filter(
+            context=context, template_name=dashboard_template,
+        )
+    except DashboardRenderStarted.RenderAlternativeDashboard as exc:
+        response = render_to_response(exc.dashboard_template, exc.template_context)
```

Comment:
It looks like, instead of raising `DashboardRenderStarted.RenderAlternativeDashboard`, a dashboard render filter could return something like this and surreptitiously change the template and/or context:

```python
return {
   "context": context, 
   "template_name": "alternative_template.html",
}
```

It's cleaner if everything is handled using the exception mechanism, even changes to the `context` dict:

```suggestion
    try:
        # .. filter_implemented_name: DashboardRenderStarted
        # .. filter_type: org.openedx.learning.dashboard.render.started.v1
        DashboardRenderStarted.run_filter(
            context=context, template_name=dashboard_template,
        )
    except DashboardRenderStarted.RenderAlternativeDashboard as exc:
        response = render_to_response(exc.dashboard_template, exc.template_context)
```

Reply:
This design was intended as the following:
Happy path -> modify return values to render a valid dashboard but modified
Unhappy path -> modify the return values through exceptions
For clarity we could change:
RenderAlternativeDashboard
For:
RenderInvalidDashboard?
Does this make sense? @pomegranited @felipemontoya

Reply:
That naming change would make things clearer @mariajgrimaldi , good suggestion. Also some docs to explain the expected usage?

Reply:
We mentioned it in the [OEP-50](https://open-edx-proposals.readthedocs.io/en/latest/architectural-decisions/oep-0050-hooks-extension-framework.html#specification):
> Filters on the other hand are passed data and are expected to return something. This something will be used by the platform code. The most common case is to be passed data and return the same of somewhat changed, but similar data. Other common case would be to receive data and raise an exception in accordance with the definition of the hook. Most likely to completely halt the process that would happen after the hook.

But it would be useful to add it in a straightforward way to the opened-filters documentation.

Again, thank you for the great suggestions! :smile:

Reply:
Ah cool, I haven't read all the OEPs :)  I'd be fine with just referencing the relevant ones in your openedx-filters README, and then leave it to the users of the library to read up on the details there. No need to re-write what's already been written and reviewed :)

Reply:
This is working great now :) But we need a small update to 

https://github.com/eduNEXT/openedx-filters-samples/blob/master/openedx_filters_samples/samples/pipeline.py#L525 -- needs to raise `DashboardRenderStarted.RenderInvalidDashboard`.